model/src/main/java/io/syndesis/model/integration/Integration.java

```diff
@@ -34,39 +34,45 @@
 @JsonDeserialize(builder = Integration.Builder.class)
 public interface Integration extends WithId<Integration>, WithTags, WithName, Serializable {
 
-    enum Status { Draft, Pending, Activated, Deactivated, Deleted}
-
     @Override
     default Kind getKind() {
         return Kind.Integration;
     }
 
-    Optional<String> getConfiguration();
+    Optional<String> getDescription();
 
-    Optional<String> getIntegrationTemplateId();
+    /**
+     * The list of versioned revisions.
+     * The items in this list should be versioned and are not meant to be mutated.
+     * @return
+     */
+    List<IntegrationRevision> getRevisions();
 
-    Optional<String> getUserId();
+    Optional<IntegrationRevision> getDraftRevision();
 
-    Optional<String> getToken();
+    Optional<Integer> getDeployedRevisionNumber();
 
-    List<User> getUsers();
+    @JsonIgnore
+    default Optional<IntegrationRevision> getDeployedRevision() {
+        return getDeployedRevisionNumber().map(i -> getRevisions()
+            .stream()
+            .filter(r -> r.getVersion().isPresent() && i.equals(r.getVersion().get()))
+            .findFirst()
+            .orElse(null));
+    }
 
-    Optional<List<Connection>> getConnections();
+    IntegrationSpec getSpec();
 
-    Optional<List<? extends Step>> getSteps();
+    Optional<String> getUserId();
 
-    Optional<String> getDescription();
+    Optional<String> getToken();
 
     Optional<String> getGitRepo();
 
-    Optional<Status> getDesiredStatus();
-
-    Optional<Status> getCurrentStatus();
+    List<User> getUsers();
```

Comment:
I don't know. Users are not introduced in this PR. They were just a chunk of code moved to an other place.